ygot/proto.go

```diff
@@ -0,0 +1,51 @@
+package ygot
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/descriptorpb"
+
+	yextpb "github.com/openconfig/ygot/proto/yext"
+	wpb "github.com/openconfig/ygot/proto/ywrapper"
+)
+
+// pathsFromProto returns, from a populated proto, a map between the YANG schema
+// path (as specified in the yext.schemapath extension) and the value populated in
+// the message.
+func pathsFromProto(p proto.Message) (map[string]interface{}, error) {
```

Comment:
Is one of the purposes of this change to use the mapping from this function as an intermediate for converting from proto messages to GoStructs? By converting the map to a set of gNMI notifications and then calling `SetNode`?

Reply:
I guess one can also add proto encoding to `SetNode`

Reply:
That's right -- we have some proto input -- which has a schema path specified in it -- the wider change will go from schema tree path to data tree path, and then form the right call to `SetNode` to set the value in the parent.

This [doc](https://docs.google.com/document/d/106Qyw9Zre0kr7Gl3Xfkl4hYQv1Ml1GlTL3XsfcvOBKM/edit) described the high level approach before the new protobuf API, where a plugin to `protoc` was required (equivalent to `protoc-gen-go`), the new API makes this much cleaner.